nrpy/infrastructures/ETLegacy/interface_ccl.py

```diff
@@ -84,14 +86,15 @@ def construct_interface_ccl(
 
 public:
 """
+    (
+        evolved_variables_list,
+        auxiliary_variables_list,
+        auxevol_variables_list,
+    ) = gri.CarpetXGridFunction.gridfunction_lists()[0:3]
     if is_evol_thorn:
-        # First, EVOL type:
-        evol_gfs = [
-            f"{gfname}GF"
-            for gfname, gf in gri.glb_gridfcs_dict.items()
-            if gf.group == "EVOL"
-        ]
-        if evol_gfs:
+        if evolved_variables_list:
+            # First, EVOL type:
+            evol_gfs = [evol_gf + "GF" for evol_gf in evolved_variables_list]
```

Comment:
Would this be cleaner?
outstr += (
    f"CCTK_REAL evol_variables type = GF Timelevels=3\n{{\n  "
    f"{', '.join(evol_gfs)}\n"
    f"""}} "Evolved gridfunctions.""""
)

Reply:
pylint (or one of them) will complain if you use `f` when you aren't actually using any variables inside the string. Personally, I feel like what you wrote is far less clear than what I have. There's way more special characters because of the f's, and I don't know that we actually gain anything by pulling the join inside the string.

Reply:
Really I just want to remove string+string+string+string stuff, and use more modern practices. You can modify my suggestion.

Reply:
fr""" is probably more appropriate here.

Reply:
Please check my modifications.